diff --git a/Makefile b/Makefile
index 8c7f3523f..b6a79529c 100644
--- a/Makefile
+++ b/Makefile
@@ -90,10 +90,12 @@ RPM_DEPENDS += libuuid-devel
 RPM_DEPENDS += mbedtls-devel
 
 ifeq ($(OS_ID),fedora)
-	RPM_DEPENDS += dnf-utils
+	ifeq ("$(wildcard /usr/bin/package-cleanup)","")
+		RPM_DEPENDS += dnf-utils
+	endif
 	RPM_DEPENDS += subunit subunit-devel
-	RPM_DEPENDS += compat-openssl10-devel
-	RPM_DEPENDS += python2-devel python34-ply
+	RPM_DEPENDS += openssl-devel
+	RPM_DEPENDS += python2-devel
 	RPM_DEPENDS += python2-virtualenv
 	RPM_DEPENDS += cmake
 	RPM_DEPENDS_GROUPS = 'C Development Tools and Libraries'
